### PR TITLE
Avoid overlapping buffers in native ChaCha20

### PR DESCRIPTION
**Pull request by jasonkatonica**
When using the ChaCha20 algorithms and an input
and output buffer overlaps we should ensure that
a copy of the input buffer is made before
encrypting or decrypting. This ensures that
OpenSSL is able to process this data. OpenSSL
does not allow for overlapping input and output
buffers when performing operations on data for the ChaCha20 algorithm.

The values returned when getting the output
size for a crypto operation were also found to be
incorrect. This update matches the logic that is
within the ChaCha20Cipher.java file for
calculating the output sizes.

Encoded key material was found to be left in
memory under the right conditions. This memory
should be zeroed to avoid a copy of the key from
residing in memory for longer than necessary.
This addition was noticed when comparing the
NativeChaCha20Cipher class to the ChaCha20Cipher
class. The method getEncodedKey now will zero out
the copy of the key before throwing an exception.

Each of these changes fix the two failing tests
reported in issue [18703](https://github.com/eclipse-openj9/openj9/issues/18703).

### CODE REVIEW

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java [hunk 1]
```diff
@@ -656,6 +643,9 @@ private static byte[] getEncodedKey(Key key) throws InvalidKeyException {
         }
         byte[] encodedKey = key.getEncoded();
         if (encodedKey == null || encodedKey.length != 32) {
+            if (encodedKey != null) {
+                Arrays.fill(encodedKey, (byte)0);
+            }
```
[the remark this answers is not in the document]
Comment (jasonkatonica) [reply closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java:651]: I diffed the files between 11 and 17 and saw the change you noticed with zeroing out the key bytes. I pushed this change using [605cb86](https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/758/commits/605cb86143d26bc4316b8a49b89e61df61650628)

[the remark this answers is not in the document]
Comment (keithc-ca) [reply closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java:651]: Thank you. Could you squash this, please?

[the remark this answers is not in the document]
Comment (jasonkatonica) [reply closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java:651]: Just squashed this update.

[the remark this answers is not in the document]
Comment (keithc-ca) [reply closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java:651]: Also, please fix the typos in the commit message and description (see my comment in https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/744).

I should have paid more attention before merging https://github.com/ibmruntimes/openj9-openjdk-jdk17/pull/323, https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/130, and https://github.com/ibmruntimes/openj9-openjdk-jdk22/pull/25.

**Comment by jasonkatonica** [comment]
Updated the description:

algorothms -> algorithms

longer then necessary -> longer than necessary

File: closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jasonkatonica) [reply closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java:651]: Updated the description and commit message for typos:

algorothms -> algorithms

longer then necessary -> longer than necessary

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins compile zlinux jdk11

**Pull request merged**